### PR TITLE
Allow customizing the underlying intrinsic element

### PR DESCRIPTION
**Pull request by gabro**
`FlexView` uses `div` as a default for the underlying component.

This PR allows changing the default to another intrinsic component. For example:

```tsx
<FlexView />
<FlexView component="form" />
```

will render

```tsx
<div ... />
<form ... />
```

### CODE REVIEW

File: src/FlexView.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ function some(array: any[], predicate: (v: any) => boolean): boolean {
   return array.filter(predicate).length > 0;
 }
 
-type DivProps = Omit<React.HTMLProps<HTMLDivElement>, "ref">;
+type ElementProps = Omit<React.HTMLProps<HTMLElement>, "ref">;
```
Comment (giogonzo) [inline src/FlexView.tsx:25]: is there an easy way to make this generic? i.e. have the correct props based on the element passed

File: src/FlexView.tsx [hunk 2]
```diff
@@ -97,7 +100,7 @@ export class FlexViewInternal extends React.Component<
       hAlignContent,
       vAlignContent,
       children,
-      column
+      column,
```
Comment (giogonzo) [inline src/FlexView.tsx:103]: maybe avoid these unrelated changes, if not too much hassle

**Review by giogonzo — COMMENTED** [review]
LGMT, just two minor requests inline

**Comment by gabro** [comment]
@giogonzo done, now we get refined types for stuff like

```tsx
<FlexView
  onClick={
    e => // the type of 'e' is 'React.MouseEvent<React.ElementType<"div">, MouseEvent>'
  }
/>

<FlexView
  component='form'
  onClick={
    e => // the type of 'e' is 'React.MouseEvent<React.ElementType<"form">, MouseEvent>'
  }
/>
```

The only downside is that while doing so I've hit a 

```
Expression produces a union type that is too complex to represent.
```

which I've promptly ts-ignored :P

**Comment by giogonzo** [comment]
@gabro ouch! Should be fine though (i.e. not have TS perf impacts on projects using the lib?)

**Review by giogonzo — APPROVED** [review]

**Comment by gabro** [comment]
I've tried using this from the tests and it seems fine (aka ts server is responsive)
